readme.md

```diff
@@ -15,834 +14,1113 @@
 
-    ```
+Only use underscore `_` name as an object member with private properties:
 
-    ```javascript
+```javascript
+var object = {
+	// Public member.
+	foo: {},
+	// Also public member.
+	bar: {},
+	// Private member and everything contained inside this object is also private.
+	_: {}
+}
+```
 
-    // 5.2.1
-    // A Practical Constructor
+Avoid using underscore `_` with private variables, methods and functions. Note that all undocumented members are private by default, so there is no need beeing explicit here.
 
-    ( function() {
-        function Ctor( foo ) {
-            this.foo = foo;
+```javascript
+// Bad
+var object = {
+	_foo: {}
+};
 
-            return this;
-        }
+// Good
+var object = {
+	foo: {}
+};
 
-        CKEDITOR.tools.extend( Ctor.prototype, {
-            getFoo: function() {
-                return this.foo;
-            };
+// Also good, but note that explicitivness is only needed for easier distinction between public and private members.
+var object = {
+	/*
+	 * @private
+	 */
+	 foo: {},
 
-            setFoo: function( val ) {
-                return ( this.foo = val );
-            };
-        } );
+	/*
+	 * Some nice description here.
+	 * @property {Object}
+	 */
+	 bar: {}
+};
+```
 
-        // Expose our constructor to the global object.
-        CKEDITOR.ctor = Ctor;
-    } )();
+#### 2.6.1.5. Configuration members
 
-    ```
+If configuration member is a part of core codebase, go with `camelCaseNaming` notation.
 
+```javascript
+CKEDITOR.config.fooBarBaz = true;
+```
 
-6. <a name="naming">Naming</a>
+If configuration member is a part of [plugin](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_plugins.html) functionality, use `pluginName_optionName` pattern consisting of mixed `camelCase` plugin and option naming separated with underscore `_`.
 
+```javascript
+CKEDITOR.config.myPlugin_myProperty = true;
+```
 
-    A. You are not a human code compiler/compressor, so don't try to be one.
+## 2.7. Faces of `this`
 
-    The following code is an example of egregious naming:
+### 2.7.1. Event listeners
 
-    ```javascript
+Keep `this` "local" to the class. This means &ndash; do not bind e.g. a function in `selection.js` file to editor instance and do not create a function which should be ran in editor context. This way you'll avoid confusion what `this` refers to.
 
-    // 6.A.1.1
-    // Example of code with poor names
+```javascript
+function Selection() {
+	var editor = this.editor;
 
-    function q(s) {
-        return document.querySelectorAll(s);
-    }
-    var i,a=[],els=q("#foo");
-    for(i=0;i<els.length;i++){a.push(els[i]);}
-    ```
+	// Bad:
+	editor.on( 'setData', someCallback, editor );
 
-    Here's the same piece of logic, but with kinder, more thoughtful naming (and a readable structure):
+	// Still bad:
+	editor.on( 'setData', CKEDITOR.tools.bind( someCallback, editor ) );
 
-    ```javascript
+	// Good:
+	editor.on( 'setData', someCallback, this );
```

Comment:
A little confusing because in "bad" examples you pass/bind editor and in "good" one you pass different object (`this` which needs `this.editor`), this seem inconsistent and a little confusing about the intention of this example.